### PR TITLE
DoS tool: generate transactions using several threads

### PR DESCRIPTION
**Pull request by KirillLykov**
#### Problem

Single threaded transactions generation is not sufficient to create a massive number of transactions.

#### Summary of Changes

This PR introduces multithreading to transaction generation.
The scheme is the following:
* Several threads are generating transactions and putting them into channel
* One Sender thread is taking these transactions and sends them

For sending, we use ConnectionCache because experiments with Clients showed less pps (might be changed in follow up PRs).

### CODE REVIEW

File: dos/src/main.rs [hunk 1]
```diff
@@ -235,7 +239,170 @@ impl TransactionGenerator {
...
+                            time_generate_ns += time;
+
+                            if res.is_err() {
+                                error_count += len;
+                            }
+                            count += len;
+                        }
+                        Ok(TransactionMsg::Exit) => {
+                            info!("Worker is done");
+                            n_alive_threads -= 1;
```
Comment (sakridge) [inline dos/src/main.rs:294]: sorry, how does this work? it decrements this local n_alive_threads? which is set to the number of worker threads, and then when it hits 0 it exits?

Comment (KirillLykov) [reply dos/src/main.rs]: It didn't work, some leftovers which I left, sorry. I've updated the code with the following mechanism:
* There are N Generator threads which generate batches of transactions (size is configurable now)
* There is 1 Sender thread which sends them
* They communicate using channel

The communication through the channel might be of two types:
* Generator puts batches to channel and sender reads them
* Sender can decide to stop the program by dropping receiver of the channel in two cases: all the transactions have been sent or timeout. This logic is used for tests, when running standalone number of iterations is 0.

Comment (sakridge) [reply dos/src/main.rs]: ok, let me take a look again

**Review by sakridge — APPROVED** [review]
lgtm

**Pull request merged**